pkg/config/validation/validation.go

```diff
@@ -1514,21 +1523,62 @@ func ValidateAuthorizationPolicy(_, _ string, msg proto.Message) error {
 }
 
 // ValidateRequestAuthentication checks that request authentication spec is well-formed.
-func ValidateRequestAuthentication(_, _ string, msg proto.Message) error {
+func ValidateRequestAuthentication(name, namespace string, msg proto.Message) error {
 	in, ok := msg.(*security_beta.RequestAuthentication)
 	if !ok {
 		return errors.New("cannot cast to RequestAuthentication")
 	}
-	// TODO(diemtvu) add more details validation.
+
 	var errs error
+	emptySelector := in.Selector == nil || len(in.Selector.MatchLabels) == 0
+	if name == constants.DefaultAuthenticationPolicyName && !emptySelector {
+		errs = appendErrors(errs, fmt.Errorf("default request authentication cannot have workload selector"))
+	} else if emptySelector && name != constants.DefaultAuthenticationPolicyName {
+		errs = appendErrors(errs,
+			fmt.Errorf("request authentication with empty workload selector must be named %q", constants.DefaultAuthenticationPolicyName))
+	}
+
+	errs = appendErrors(errs, validateWorkloadSelector(in.Selector))
+
 	for _, rule := range in.JwtRules {
-		if len(rule.Issuer) == 0 {
-			errs = appendErrors(errs, fmt.Errorf("issuer must be set"))
-		}
+		errs = appendErrors(errs, validateJwtRule(rule))
 	}
 	return errs
 }
 
+func validateJwtRule(rule *security_beta.JWT) (errs error) {
+	if rule == nil {
+		return nil
+	}
+	if len(rule.Issuer) == 0 {
+		errs = multierror.Append(errs, errors.New("issuer must be set"))
+	}
+	for _, audience := range rule.Audiences {
+		if len(audience) == 0 {
+			errs = multierror.Append(errs, errors.New("audience must be non-empty string"))
+		}
+	}
+
+	if len(rule.JwksUri) != 0 {
+		if _, err := security.ParseJwksURI(rule.JwksUri); err != nil {
+			errs = multierror.Append(errs, err)
+		}
+	}
+
+	for _, location := range rule.FromHeaders {
+		if len(location.Name) == 0 {
```

Comment:
nit: location could be nil I think

Reply:
`FromHeaders` is a list parsed from yaml, it should not insert a `nil` object, right?

Reply:
Oh I thought it was a map. I think you can do: `[nil]` or something similar. but its a pretty obscure case

Reply:
It's a list of {name, prefix} tuple. It's actually have the [`[(google.api.field_behavior) = REQUIRED]` ](https://github.com/istio/api/blob/master/security/v1beta1/jwt.proto#L135)annotation so not sure if I still need to do this. (I clone these from the old API, btw)